### DISCUSSION TITLE
img2img/inpainting window fails to load BIG filesize images

### DISCUSSION THREAD

**Discussion by YourFriendlyNeighborhoodMONKE** [vladmandic/sdnext#2387] [Q&A] [answered]
Hello!

When you input an image that is larger than approximately 22-24 megabytes in file size, the image does not load into the img2img or inpainting window

Automatic1111 has this same issue and it _might_ be Gradio related

- The issue is **NOT** about image pixel dimensions being large as you can compress a huge 10k+ pixel images without problems and have it working just fine

This makes it practically impossible to do inpainting on large filesize images, which is crucial in my workflow

Has anyone been able to find a workaround for this issue
OR know how to modify the code as there could be some artificial limitation since the filesize limit seems to be so exact?

**Comment by vladmandic** [comment]
just to double-check, do you have any display scaling set on your desktop?

**Reply by YourFriendlyNeighborhoodMONKE** [reply]
No.

My display scaling setting is set to 100%

**Comment by YourFriendlyNeighborhoodMONKE** [comment]
Please don't forget MONKE

Very easy to test with image over 24mb - It won't load on img2img or inpainting

Easiest way to find a PNG that large is simply make one by resizing any existing image you have

5000x5000+ as a PNG should about do it

**Reply by Tillerz** [reply]
Have you tried to use a .WEBP version of a 24 MB PNG instead, just to see if that works? Just to see if that bugs out, too.

**Reply by YourFriendlyNeighborhoodMONKE** [reply, marked as answer]
I just tested .webp as you suggested (I've also tested with JPG before)
As expected 22.7 megabytes still worked, but 27mb did not

I'm very glad however that you asked since uncompressed .webp takes so much less space than PNG
I always ignored it due to bad experiences with downloaded images 

Turns out **.webp is a great workaround for my problem!**
The 22.7mb file is 113 megapixels and it's pretty much the highest pixel count I pratically ever need

So thank you for the suggestion 👍 



-------------------


On opening gradio issue as vladmandic suggested:

I opened an issue at gradio yesterday:
https://github.com/gradio-app/gradio/issues/6081

Someone speculated it might be from gradio using base64 and the issue might be resolved once A1111 updates to a newer gradio
He mentioned "version 4", which was unclear whether he meant 3.4 or 4.x
As far as I know A1111 currently uses 3.35.x

Not sure thus if the latest gradio has changed this aspect and whether A111/vladmandic could be upgraded
Let's see if there's more information coming

**Reply by vladmandic** [reply]
v4 referers to actual 4.0 which is in the making. i have several pretty big issues with gradio inpainting that gradio openly stated that they are real issues, but they will not even try to fix them since v4 will completely reimplement inpainting from scratch.

but another issue is that gradio is breaking backwards compatibility badly, so upgrading to new version will be extremely painful and will take a long time.

**Reply by YourFriendlyNeighborhoodMONKE** [reply]
Thanks for the clarification!

Yeah, I can definitely imagine how big of a nightmare such a drastic part of everything that breaks compatibility could be... Yikes!

I'll mark this conversation as an answer in the sense of a _practical workaround_ for the issue for the time being 👍 

Big thanks to you both!

**Comment by vladmandic** [comment]
reposting on an issue only 2days old only makes it LESS likely to be prioritized.
and this is a discussion item, not even filed as an issue.

also, like you said:
> Automatic1111 has this same issue and it might be Gradio related

so really not much that can be done - i suggest to reproduce with a basic gradio install and file issue in their git repo.

**Reply by YourFriendlyNeighborhoodMONKE** [reply]
I apologize for behaving in a rude way!

I'm pretty new overall to the proper GitHub etiquette

Your suggestion is sound
